crawl-ref/source/dat/des/serial/rivers.des

```diff
@@ -0,0 +1,168 @@
+# A serial layout focusing on little streams appearing through the
+# dungeon. Similar to bayou, but with less vegetation.
+
+NAME:   serial_rivers
+TAGS:   allow_dup extra luniq_serial decor nolayout_encompass
+DEPTH:  D, Depths, Lair
+CHANCE: 1%
+{{
+  local c = 3 + crawl.random2(3)*crawl.random2(2)
+  hook("post_place", function()
+    dgn.place_maps{tag="serial_rivers", count=c}
+  end)
+}}
+
+NAME:   chequers_serial_rivers_micro_a
+TAGS:   serial_rivers allow_dup extra decor no_pool_fixup transparent
+DEPTH:  D, Depths, Lair
+MAP
+    @
+xxx...xxx
+xwwwWWwwx
+xxx...xxx
+    @
+ENDMAP
+
+NAME:   chequers_serial_rivers_micro_b
+TAGS:   serial_rivers allow_dup extra decor no_pool_fixup transparent
+DEPTH:  D, Depths, Lair
+MAP
+    @
+xxx...x
+xwwwWWx
+xxx...x
+    @
+ENDMAP
+
+NAME:   chequers_serial_rivers_room_a
+TAGS:   serial_rivers allow_dup extra decor no_pool_fixup transparent
+DEPTH:  D, Depths, Lair
+MAP
+xxxxxx+
+xw....x
+xw....x
+xw....x
+xwwwwwx
+xxxxxxx
+ENDMAP
+
+NAME:   chequers_serial_rivers_room_b
+TAGS:   serial_rivers allow_dup extra decor transparent
+DEPTH:  D, Depths, Lair
+SUBST:  w = W:5 w
+MAP
+xxx+xxx
+xwwWwwx
+xwWwWwx
+xWwWwWx
+xwWwWwx
+xxx+xxx
+ENDMAP
+
+NAME:   chequers_serial_rivers_small_a
+TAGS:   serial_rivers allow_dup extra decor transparent
+DEPTH:  D, Depths, Lair
+MAP
+  w
+ www
+  wwww..w@
+@...WWwwww
+         www
+ENDMAP
+
+NAME:   chequers_serial_rivers_small_b
+TAGS:   serial_rivers allow_dup extra decor transparent
+DEPTH:  D, Depths, Lair
+KPROP:  w = no_tele_into
+MAP
+@     w
+.ww   w
+ .ww. Www.
+.  Ww. ..wW@
+ .      ww
+  .   w
+ .  www
+w.
+ww@
+w .
+ENDMAP
+
+NAME:   chequers_serial_rivers_marsh
+TAGS:   serial_rivers allow_dup extra decor transparent
+DEPTH:  D, Depths, Lair
+SUBST:  . = w. t:2
+MAP
+     @
+ ........ 
+..........
+   .......
+..........
+.... ...  
+..........@
+   ...... 
+@.......  
+.......   
+ ........ 
+    @
+ENDMAP
+
+NAME:   chequers_serial_rivers_small_3
+TAGS:   serial_rivers allow_dup extra decor no_pool_fixup transparent
+DEPTH:  D, Depths, Lair
+SUBST:  W = .:3 w:2 W, . = . W:4
+MAP
+       .......... 
+      ............
+xxxx@..............xxxx
+xwwwwwWWWWWWWWWWWwwwwwx
+xwwwwwWWWWWWWWWWWwwwwwx
+xxxx..............@xxxx
+       .......... 
+ENDMAP
+
+NAME:   chequers_serial_rivers_large_a
+TAGS:   serial_rivers allow_dup extra decor no_pool_fixup transparent
+DEPTH:  D, Depths, Lair
+SUBST:  W = .:4 w:4 W, - = W .
```

Comment:
You're going to have to do a more careful SUBST here, since you will (rarely) create little shallow water teleport closets where deep water surrounds a central shallow water or floor tile. It would probably be a good idea to have a central one-tile region of shallow water with a guaranteed shallow-water connection to the `-` outer region. This will still allow plenty of squares to have randomized water but will guarantee the player cannot be trapped.